checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleExecutor.java

```diff
@@ -92,25 +101,27 @@ private void executeWithClassLoader(URLClassLoader projectClassloader) {
             checker.addListener(listener);
 
             final File xmlReport = configuration.getTargetXmlReport();
-            if (xmlReport != null) {
+            if (Objects.nonNull(xmlReport)) {
                 LOG.info("Checkstyle output report: {}", xmlReport.getAbsolutePath());
                 xmlOutput = FileUtils.openOutputStream(xmlReport);
-                checker.addListener(new XMLLogger(xmlOutput, true));
+                checker.addListener(new XMLLogger(xmlOutput,
+                        AutomaticBean.OutputStreamOptions.CLOSE));
```

Comment:
this is not related to Sonar update, but good catch, please move to separate commit/PR for quick merge.